frontend/src/components/Admin/EditTeamEvents.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import { Form, Radio, Card } from 'semantic-ui-react';
+import { Emitters } from '../../utils';
+import { Member } from '../../API/MembersAPI';
+import TeamEventDetails from './TeamEventDetails';
+
+type TeamEvent = {
+  name: string;
+  date: string;
+  numCredits: string;
+  hasHours: boolean;
+  membersPending: Member[];
+  membersApproved: Member[];
+};
+
+const EditTeamEvents: React.FC = () => {
+  const [teamEventName, setTeamEventName] = useState('');
+  const [teamEventDate, setTeamEventDate] = useState('');
+  const [teamEventCreditNum, setTeamEventCreditNum] = useState('');
+  const [teamEventHasHours, setTeamEventHasHours] = useState(false);
+
+  const teamEvents: TeamEvent[] = [
+    {
+      name: 'Coffee Chat',
+      date: 'Sept 3',
+      numCredits: '0.5',
+      hasHours: false,
+      membersPending: [],
+      membersApproved: []
+    },
+    {
+      name: 'Club Fest',
+      date: 'Sept 5',
+      numCredits: '0.5',
+      hasHours: true,
+      membersPending: [],
+      membersApproved: []
+    }
+  ];
+
+  const createTeamEvent = (teamEvent: TeamEvent) => {
+    // send new event to backend
+  };
+
+  const submitTeamEvent = () => {
+    if (!teamEventName) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Event Name',
+        contentMsg: 'Please enter the name of the team event!'
+      });
+    } else if (!teamEventDate) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Event Date',
+        contentMsg: 'Please enter the date of the team event!'
+      });
+    } else if (
+      !teamEventCreditNum ||
+      teamEventCreditNum === '' ||
+      isNaN(Number(teamEventCreditNum)) ||
+      Number(teamEventCreditNum) < 1
+    ) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Event Credit Amount',
+        contentMsg: 'Please enter how many credits the event is worth!'
+      });
+    } else {
+      const newTeamEvent: TeamEvent = {
+        name: teamEventName,
+        date: teamEventDate,
+        numCredits: teamEventCreditNum,
+        hasHours: teamEventHasHours,
+        membersPending: [],
+        membersApproved: []
+      };
+      createTeamEvent(newTeamEvent);
+      Emitters.generalSuccess.emit({
+        headerMsg: 'Team Event Created!',
+        contentMsg: 'The team event was successfully created!'
+      });
+    }
+  };
+
+  return (
+    <div>
+      <Form
+        style={{
+          width: '60%',
+          alignSelf: 'center',
+          margin: 'auto',
+          padding: '3rem 0 5rem 0'
+        }}
+      >
```

Comment:
If it is easier, we can also keep styles in a separate css file. What do you think?